src/instructlab/sdg/datamixing.py

```diff
@@ -29,13 +30,20 @@ def _adjust_train_sample_size(ds: Dataset, num_samples: int):
 
 def _load_ds(path, sampling_size, num_proc):
     """
-    Load a dataset from the given file path and select sampling_size
+    Load a dataset from the given location and select sampling_size
     number/ratio of samples from it, ensuring the loaded dataset has only
     ALLOWED_COLS columns in it with any additional columns moved to the
     metadata section.
+
+    If a jsonl file is given, we load the dataset from disk. Otherwise, we
+    load the path given from HuggingFace.
     """
-    logger.info(f"Loading dataset from {path} ...")
-    dataset = load_dataset("json", data_files=path, split="train")
+    if path.endswith(".jsonl"):
+        logger.info(f"Loading dataset from {path} ...")
+        dataset = load_dataset("json", data_files=path, split="train")
```

Comment:
You know, it's probably important we support relative paths here too?

e.g. if our default recipe referenced a `.jsonl` file as a dataset, we'd need it to be relative to some install path

Reply:
We might be able to assume these are always absolute paths at the moment, as the recipes are written as absolute paths and any downstream usage of this I've heard assumed that these files would be in a known absolute path. But you bring up a good point that we should test that assumption, as well as an overall point about portability of the recipe file if the paths to jsonl files aren't relative to the data generation output directory, at least for the generated recipe yaml files and outputs.

Reply:
I had some offline discussion with the team and we agreed it likely makes more sense for these to be relative paths, relative to the generation output directory. So, my example recipe output above would change from:

```yaml
datasets:
- path: instructlab/InstructLabCommunity
  sampling_size: 1.0
- path: /home/my_user/.local/share/instructlab/datasets/node_datasets_2024-07-24T15_07_58/knowledge_tonsils_overview_e2e-tonsils_p10.jsonl
  sampling_size: 1.0
sys_prompt: "I am, Red Hat\xAE Instruct Model based on Granite 7B, an AI language\
  \ model developed by Red Hat and IBM Research, based on the Granite-7b-base language\
  \ model. My primary function is to be a chat assistant."
```
to:

```yaml
datasets:
- path: instructlab/InstructLabCommunity
  sampling_size: 1.0
- path: node_datasets_2024-07-24T15_07_58/knowledge_tonsils_overview_e2e-tonsils_p10.jsonl
  sampling_size: 1.0
sys_prompt: "I am, Red Hat\xAE Instruct Model based on Granite 7B, an AI language\
  \ model developed by Red Hat and IBM Research, based on the Granite-7b-base language\
  \ model. My primary function is to be a chat assistant."
```

For robustness sake, they should probably accept either absolute or relative with the relative option becoming relative to the output directory.

Reply:
Loading datasets from paths relative to the recipe file is now supported.